packages/mdc-dom/ponyfill.ts

```diff
@@ -47,3 +47,29 @@ export function matches(element: Element, selector: string): boolean {
       || element.msMatchesSelector;
   return nativeMatches.call(element, selector);
 }
+
+/**
+ * Used to compute the estimated scroll width of elements. When an element is
+ * hidden due to display: none; being applied to a parent element, the width is
+ * returned as 0. However, the element will have a true width once no longer
+ * inside a display: none context. This method computes an estimated width when
+ * the element is hidden or returns the true width when the element is visble.
+ * @param {HTMLElement} element the element whose width to estimate
+ */
+export function estimateScrollWidth(element: HTMLElement): number {
+  // Check the offsetParent. If the element inherits display: none from any
+  // parent, the offsetParent property will be null (see
+  // https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent).
+  // This check ensures we only clone the node when necessary.
+  if (element.offsetParent !== null) {
+    return element.scrollWidth;
+  }
+
+  const clone = element.cloneNode(true) as HTMLElement;
+  clone.style.setProperty('position', 'absolute');
+  clone.style.setProperty('transform', '-9999px, -9999px');
```

Comment:
Does this need `translate(...)` in the string around the value?

Reply:
Sure does! Thanks Liz